RegularExpressionAndExceptionHandling/RegularExpressionAndExceptionHandling/ValidPhoneNumber.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Text.RegularExpressions;
+
+namespace RegularExpressionAndExceptionHandling
+{
+    /// <summary>
+    /// ValidPhoneNumber class uses a regular expression to validate all valid forms of United States phone numbers.
+    /// It passes in a string, phoneNumber, to validate against a pattern. 
+    /// </summary>
+    public static class ValidPhoneNumber
+    {
+        public static bool PhoneNumber(string phoneNumber)
+        {
+           return Regex.IsMatch(phoneNumber, @"^(?:\+?(1))?[-./ (]*(\d{3})?[-./ )]*(\d{3})[-./ ]*(\d{4})*$");
```

Comment:
My test set of phone numbers identified a few gaps when detecting invalid phone numbers. The following should have been considered invalid but were not:

(444 333-2222
444) 333-2222
24443332222
 555-333-2222 (note the whitespace at the beginning of this one)

(-1.0)